src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -290,70 +295,105 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
+        }
 
-      let estimatedFees
+        const amount =
+          fieldName === SendFormFields.FiatAmount
+            ? bnOrZero(bn(inputValue).div(price)).toString()
+            : bnOrZero(bn(inputValue).times(price)).toString()
+
+        setValue(key, amount)
+
+        // TODO: turn to actual validators
+        let estimatedFees
+        let hasEnoughNativeTokenForGas = false
+        let hasValidBalance = false
+
+        try {
+          setLoading(true)
+          estimatedFees = await debouncedEstimateFormFees()
+        } catch (e) {
+          throw new Error('common.insufficientFunds')
+        } finally {
+          setLoading(false)
+        }
 
-      try {
-        estimatedFees = await estimateFormFees()
-        setValue(SendFormFields.EstimatedFees, estimatedFees)
-      } catch (e) {
-        setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
-        setLoading(false)
+        if (estimatedFees === undefined) {
+          throw new Error('common.generalError')
+        }
 
-        throw e
-      }
+        if (estimatedFees instanceof Error) {
+          throw estimatedFees.message
+        }
+
+        hasEnoughNativeTokenForGas = nativeAssetBalance.minus(estimatedFees.fast.txFee).isPositive()
+
+        const { cryptoAmount } = getValues()
+        hasValidBalance = cryptoHumanBalance.gte(cryptoAmount)
+
+        if (!hasValidBalance) {
+          throw new Error('common.insufficientFunds')
+        }
+
+        if (!hasEnoughNativeTokenForGas) {
+          setValue(SendFormFields.AmountFieldError, [
+            'modals.send.errors.notEnoughNativeToken',
+            { asset: feeAsset.symbol },
+          ])
+          setLoading(false)
+          return
+        }
 
-      const values = getValues()
-
-      const hasValidBalance = cryptoHumanBalance.gte(values.cryptoAmount)
-      const hasEnoughNativeTokenForGas = nativeAssetBalance
-        .minus(estimatedFees.fast.txFee)
-        .isPositive()
-
-      if (!hasValidBalance) {
-        setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
-      } else if (!hasEnoughNativeTokenForGas) {
-        setValue(SendFormFields.AmountFieldError, [
-          'modals.send.errors.notEnoughNativeToken',
-          { asset: feeAsset.symbol },
-        ])
-      } else {
         // Remove existing error messages because the send amount is valid
-        setValue(SendFormFields.AmountFieldError, '')
+        if (estimatedFees !== undefined) {
+          setValue(SendFormFields.AmountFieldError, '')
+          setValue(SendFormFields.EstimatedFees, estimatedFees)
+        }
+      } catch (e) {
+        if (e instanceof Error) {
+          setValue(SendFormFields.AmountFieldError, e.message)
+        }
+        // setValue(SendFormFields.AmountFieldError, 'common.insufficientFunds')
+        // TODO: useForm::setError instead of setValue
+        // setError(SendFormFields[key], { type: 'custom', message: 'Insufficient Funds' })
```

Comment:
Can we either `setError` here or remove this comment and partial implementation?

Reply:
For context, this is how we do forms error handling across web and while this is not the standard react-hook-form way to do it, it's consistent so I'm happier with that implementation and an issue to change it everywhere, than inconsistencies in implementations